Add pod2pod strict mode for WireGuard (#21856)
cilium-team/cilium: Documentation/security/network/encryption.rst

endpoint on a remote node, it will start encrypting packets using the encryption
key of the remote node.

One workaround for this issue is to ensure that the endpoint is not allowed to
send unencrypted traffic to arbitrary targets outside of the cluster. This can
be achieved by defining an egress policy which either completely disallows
traffic to ``reserved:world`` identities, or only allows egress traffic
to addresses outside of the cluster to a certain subset of trusted IP
addresses using ``toCIDR``, ``toCIDRSet`` and ``toFQDN`` rules.
See :ref:`policy_examples` for more details about how to write network
policies that restrict egress traffic to certain endpoints.

Another way to mitigate this issue is to set ``encryption.strictMode.enabled``
to ``true`` and the expected pod CIDR as ``encryption.strictMode.cidr``.
This encryption strict mode enforces that traffic exiting a node
to the set CIDR is always encrypted. The tradeoff being that the information

> about new pod endpoints need to propagate to the node before it can send out

Reviewer: s/need/needs/

> traffic to it.

Reviewer: hmm, should it be "to them" (referring to the new endpoints) ?

Enabling encryption strict mode currently has the following limitations:

 - Only WireGuard encryption is supported.

>  - The pod CIDR must to be separate from the node CIDR. The easiest way to

Reviewer: "must to be separate" sounds like should be reworded, perhaps "must be separate", or "must be different from" ?

>    achieve this, is to use VXLAN routing.

Reviewer 2: We should explicitly mention here that strict-mode encryption won't work with the AKS, EKS, and AlibabaCloud IPAMs.

endpoint on a remote node, it will start encrypting packets using the encryption
key of the remote node.

> One workaround for this issue is to ensure that the endpoint is not allowed to

Reviewer 3: Nice. ✨

send unencrypted traffic to arbitrary targets outside of the cluster. This can
be achieved by defining an egress policy which either completely disallows
traffic to ``reserved:world`` identities, or only allows egress traffic
to addresses outside of the cluster to a certain subset of trusted IP
addresses using ``toCIDR``, ``toCIDRSet`` and ``toFQDN`` rules.
See :ref:`policy_examples` for more details about how to write network
policies that restrict egress traffic to certain endpoints.

Another way to mitigate this issue is to set ``encryption.strictMode.enabled``
to ``true`` and the expected pod CIDR as ``encryption.strictMode.cidr``.
This encryption strict mode enforces that traffic exiting a node

> to the set CIDR is always encrypted. The trade-off being that the information
> about new pod endpoints needs to propagate to the node before it can send out
> traffic to them.

Reviewer 3: ```suggestion
to the set CIDR is always encrypted. Be aware that information
about new pod endpoints must propagate to the node before the node can send
traffic to them.
```

> Enabling encryption strict mode currently has the following limitations:

Reviewer 3: Clarity nit. Avoid statements about the future (or that point to it indirectly).
```suggestion
Encryption strict mode has the following limitations:
```

- Only WireGuard encryption is supported.
- The pod CIDR and therefore the encryption strict mode CIDR must be IPv4.
- To disable all dynamic lookups, you must use direct routing mode and the
  node CIDR and pod CIDR must not overlap. Otherwise,
  ``encryption.strictMode.allowDynamicLookup`` must be set to ``true``.

>   This allows unencrypted traffic which is send from or to an IP address
>   associated with a node identity.

Reviewer 3: Small nits for clarity
```suggestion
  This allows unencrypted traffic sent from or to an IP address
  associated with a node identity.
```
